### PR TITLE
Use prom-client to implement metrics API

### PR DESCRIPTION
**Pull request by paf31**
We weren't implementing the metrics API at all before.

### CODE REVIEW

**Review by BenoitRanque — APPROVED** [review]
This looks good!

File: package.json [hunk 1]
```diff
@@ -21,22 +21,23 @@
...
+    "@opentelemetry/semantic-conventions": "^1.24.0",
     "commander": "^11.0.0",
     "fastify": "^4.23.2",
-    "pino-pretty": "^10.2.3"
+    "pino-pretty": "^10.2.3",
+    "prom-client": "^15.1.2"
   },
   "devDependencies": {
-    "@tsconfig/node18": "^18.2.2",
+    "@tsconfig/node20": "^20.1.4",
```
Comment (daniel-chambers) [inline package.json:40]: Why did this change? We're still on Node 18 in this project as defined in the .nvmrc file, and the node types below are also at node 18.

I would revert this unless you're intending on upgrading the project to be built on Node 20.

**Comment by daniel-chambers** [comment]
Also would be good to update the changelog.md file with a quick summary.

**Review by BenoitRanque — APPROVED** [review]
This looks good!

**Pull request merged**